### PR TITLE
Rev 'build-tools'

### PR DESCRIPTION
**Pull request by DLehenbauer**
Rev `@fluidframework/build-tools` to ^0.2.29883 to pick up new package-lock.json verification to ensure commits use lockfile version 1 (i.e., not NPM 7).

### CODE REVIEW

File: .devcontainer/.zshrc [hunk 1]
```diff
@@ -110,25 +110,31 @@ if [ "$(stat -c '%U' /usr/local/share/npm-global)" != "node" ]; then sudo chown
```
Comment (DLehenbauer) [inline .devcontainer/.zshrc:110]: This first file is an orthogonal change that only effects developers using a VS Code Dev Container (AFAIK, that's just me.)

The change puts me on the same version of 'fluid-build' that everyone else is using instead of building/using a private fluid-build.

The motivation is that it will hopefully help me to remember to deploy 'fluid-build' after making changes.

File: server/historian/package.json [hunk 2]
```diff
@@ -44,7 +44,7 @@
     "temp-directory": "nyc/.nyc_output"
   },
   "devDependencies": {
-    "@fluidframework/build-tools": "^0.2.25384",
```
Comment (curtisman) [inline server/historian/package.json:47]: does the lock file needs to be updated here as well?

**Review by curtisman — APPROVED** [review]
:shipit:

File: server/historian/package.json (hunk 2 above)
Comment (DLehenbauer) [reply server/historian/package.json:47]: Huh.  Never realized 'Historian' isn't included in 'fluid-build --all --install'.  Thanks for pointing that out.

Looks like the Historian 'package-lock.json' is pretty stale.  Maybe I'll do that one in a follow up commit so it's easy to pinpoint if the minor version bumps cause trouble.

Comment (DLehenbauer) [reply server/historian/package.json:47]: Followup here: #6767
